src/Microsoft.SqlTools.ServiceLayer/EditData/Session.cs

```diff
@@ -0,0 +1,194 @@
+﻿//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+using System;
+using System.Collections.Concurrent;
+using System.IO;
+using System.Linq;
+using Microsoft.SqlTools.ServiceLayer.EditData.Contracts;
+using Microsoft.SqlTools.ServiceLayer.EditData.UpdateManagement;
+using Microsoft.SqlTools.ServiceLayer.QueryExecution;
+using Microsoft.SqlTools.ServiceLayer.Utility;
+
+namespace Microsoft.SqlTools.ServiceLayer.EditData
+{
+    public class Session
```

Comment:
It's easier to validate that the correct behavior occurred and to fake improper behavior in unit tests if I make these properties internal. Since they're meant to only be used outside the class in unit tests, I feel like they should stay as member variables, but I'll make them properties as per your request.